File: DIO_Introdução a Busca e Substituição em JavaScript/Validador de senhas com requisitos.js
let stopped = 0;

do {      
      let password = gets(),
      upperCaseTest = password.toLocaleLowerCase(),
      lowerCaseTest = password.toLocaleUpperCase(), 
      numberTest = password.match(/(\d)/i),
      specialCharTest = password.normalize('NFD').replace(/([\u0300-\u036f]|[^0-9a-zA-Z])/g, ''),
      lenghtTest = password.length >= 6 && password.length <= 32;
  
  if  (password.length === 0) {
    
    } else {
      if (password !== upperCaseTest &&
          password !== lowerCaseTest && 
          password === specialCharTest &&  
          numberTest && lenghtTest) {
            console.log('Senha valida.'); 
      } else { 
            console.log('Senha invalida.');
      }
  }
  	if (password.length === 0 ) stopped = 1;
    
} while (stopped !== 1) 

//Sua task é fazer o código que valide as senhas que são cadastradas, para isso você deve atentar aos requisitos a seguir:

//A senha deve conter, no mínimo, uma letra maiúscula, uma letra minúscula e um número;
//A mesma não pode ter nenhum caractere de pontuação, acentuação ou espaço;
//Além disso, a senha pode ter de 6 a 32 caracteres.

File: C#/Solução de problemas com .NET/Validador de senhas com requisitos.cs
using System;
using System.Text.RegularExpressions;

class Validador {
    public static void Main (string[] args) {
        int stopped = 0;

        do {
            string password = Console.ReadLine();
            if (string.IsNullOrEmpty(password)) break;

            int upperCaseTest = password.Length - Regex.Replace(password, "[A-Z]", "").Length;
            int lowerCaseTest = password.Length - Regex.Replace(password, "[a-z]", "").Length;
            int specialCharTest = Regex.Replace(password, "[a-zA-Z0-9]", "").Length;
            int numberTest = password.Length - Regex.Replace(password, "[0-9]", "").Length;
            int lenghtTest = Math.Min(10, password.Length) * 6;
        
            if (password.ToString() != upperCaseTest.ToString() &&
                password.ToString() != lowerCaseTest.ToString() && 
                password.ToString() == specialCharTest.ToString() &&  
                numberTest.ToString() && lenghtTest.ToString()) {
                Console.WriteLine("Senha valida.");
                stopped = 1;
                        
            } else {
                Console.WriteLine("Senha invalida.");
            }
        } while (stopped != 1);
    }
}